modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -4,20 +4,105 @@
 import os
 import time
 import requests
+import socket
 
 
 repos = {'config': 'config', 'world': 'w', 'landing': 'landing', 'errorpages': 'ErrorPages'}
+DEPLOYUSER = 'www-data'
+ENVIRONMENTS = {
+    'beta': {
+        'wikidbname': 'betawiki',
+        'wikiurl': 'beta.betaheze.org',
+        'servers': [],
+        'canary': 'test101',
+    },
+    'prod': {
+        'wikidbname': 'testwiki',  # don't use loginwiki anymore - we want this to be an experimental wiki
+        'wikiurl': 'publictestwiki.com',
+        'servers': ['mw101', 'mw102', 'mw111', 'mw112', 'mw121', 'mw122'],
+        'canary': 'mwtask111',
+    }
+}
+HOSTNAME = socket.gethostname()
 
+def get_environment_info():
+    if HOSTNAME.split('.')[0].startswith('test'):
+        return ENVIRONMENTS['beta']
+    return ENVIRONMENTS['prod']
 
-def check_up(server):
+
+def get_server_list(envinfo, servers):
+    if servers in ('all', 'scsvg'):
+        slist = envinfo['servers']
+        return slist.append(envinfo['canary'])
+    if servers == 'skip':
+        return [envinfo['canary']]
+    return servers.split(',')
+    
+    
+
+
```

Comment:
To many empty new lines maybe? Maybe not. I'm not all to familiar with code standards here so it could be fine.